src/plugins/template/tests/functional/shared_tests_instances/behavior/ov_executable_network/exec_network_base.cpp

```diff
@@ -24,4 +35,321 @@ INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests,
                                             ::testing::ValuesIn(configs)),
                          OVCompiledModelBaseTestOptional::getTestCaseName);
 
+namespace {
+using op::v0::Parameter, op::v0::Constant, op::v1::Add;
+
+std::shared_ptr<Model> make_model_with_weights(const ov::Tensor& w) {
+    constexpr auto precision = element::f32;
+
+    auto weights = std::make_shared<Constant>(w);
+    auto input = std::make_shared<Parameter>(precision, Shape{1});
+    auto add = std::make_shared<Add>(input, weights);
+
+    weights->set_friendly_name("weights");
+    input->set_friendly_name("input");
+    add->set_friendly_name("add");
+
+    auto model = std::make_shared<Model>(OutputVector{add}, ParameterVector{input}, "Simple with weights");
+    util::set_tensors_names(AUTO, *model, {}, {{0, {"add"}}});
+    return model;
+}
+
+std::shared_ptr<Model> make_model_with_weights() {
+    constexpr auto precision = element::f32;
+
+    auto weights = std::make_shared<Constant>(element::f32, Shape{5}, std::vector<float>{1.0f});
+    auto input = std::make_shared<Parameter>(precision, Shape{1});
+    auto add = std::make_shared<Add>(input, weights);
+
+    weights->set_friendly_name("weights");
+    input->set_friendly_name("input");
+    add->set_friendly_name("add");
+
+    auto model = std::make_shared<Model>(OutputVector{add}, ParameterVector{input}, "Simple with weights");
+    util::set_tensors_names(AUTO, *model, {}, {{0, {"add"}}});
+    return model;
+}
 }  // namespace
+
+TEST_P(OVCompiledModelBaseTest, import_from_weightless_blob) {
+    const auto w_file_path =
+        ov::util::path_join({utils::getCurrentWorkingDir(), utils::generateTestFilePrefix() + "_weights.bin"});
+
+    std::stringstream export_stream;
+    {
+        configuration.emplace(ov::cache_mode(ov::CacheMode::OPTIMIZE_SIZE));
+        auto model = make_model_with_weights();
+        auto compiled_model = core->compile_model(model, target_device, configuration);
+        ASSERT_FALSE(!compiled_model);
+        compiled_model.export_model(export_stream);
```

Comment:
should we also check that `compiled_model` has property cache mode set to `OPTIMIZE_SIZE`?

the same for `compile_from_weightless_blob` test